src/ex1/script.js

```diff
@@ -0,0 +1,122 @@
+const addBtn = document.querySelectorAll('.add-task-btn');
+const clearAllBtn = document.querySelectorAll('.clear-all-btn');
+const addTaskField = document.querySelectorAll('.add-task-field');
+
+
+const taskList = [];
+addChillMsg();
+updateFooter();
+
+
+
+addBtn[0].addEventListener('click', () => {
+    addTaskFunc();
+});
+
+addTaskField[0].addEventListener("keydown", (event) => {
+    if (event.key === "Enter") {
+        addTaskFunc();
+    }
+});
+
+clearAllBtn[0].addEventListener('click', () => {
+    const tasks = document.querySelectorAll('.tasks');
+    tasks[0].replaceChildren();
+    while (taskList.length) taskList.pop();
+    tasks[0].append(ChillMsg());
+    updateFooter();
+});
+
+function addTaskFunc() {
+    const newTaskField = document.querySelectorAll('.add-task-field');
+    const taskText = newTaskField[0].value;
+    const tasks = document.querySelectorAll('.tasks');
+    console.log(taskText);
+    if (taskText == "")
+        alert("Enter new task!");
+    else if (taskList.indexOf(taskText) !== -1) {
+        alert("Task already exist!\nEnter new task!")
+    }
+    else {
+        if (taskList.length === 0) tasks[0].replaceChildren();
+        tasks[0].append(newTaskElement(taskText));
+    }
+    resetInputField(newTaskField);
+    updateFooter();
+}
+
+function resetInputField(newTaskField) {
+    newTaskField[0].value = "";
+    newTaskField[0].focus();
+}
+
+
+function newTaskElement(text) {
+    taskList.push(text);
+    const taskDiv = document.createElement('div');
+    taskDiv.classList.add('task');
+    const h3 = document.createElement('h3');
+    h3.classList.add('task-content');
+    h3.textContent = text;
+    taskDiv.append(h3);
+    const span = document.createElement('span');
+    const delBtn = document.createElement('i');
+    delBtn.classList.add('fa-solid');
+    delBtn.classList.add('fa-trash');
+    delBtn.classList.add('fa-1x');
+
+    delBtn.onclick = function () {
+        const getTaskContent = this.parentNode.parentNode.getElementsByTagName('h3')[0].textContent;
```

Comment:
this is usually a bad practice, if you'll want to change the UI a bit and add another element you'll need to add another parent or something, basically this function is very coupled with the dom
you can solve it using closest function

Reply:
I used the closest function to get the div element instead of '.parentNode.parentNode' and then did the same.
Not sure if that is what you meant.